src/Terrain/Location.java

```diff
@@ -2,13 +2,15 @@
 
 import Units.Unit;
 import CommandingOfficers.Commander;
+import Engine.XYCoord;
 
 public class Location
 {
 
   private Environment environs = null;
   private Commander owner = null;
   private Unit resident = null;
+  private final XYCoord coords;
```

Comment:
I'm not sure I want to add the coordinates as a member of the `Location` class; that already exists in the GameMap, and it's not good to repeat data when it's possible to avoid it. Could the Commander just store a list of `XYCoord`s instead of a list of `Location`s?
I know Unit has the same problem, but there didn't seem to be a good way to avoid it at the time, and there is some extra bookkeeping as a result.

Reply:
It is a good idea to avoid duplication of data, but I feel like the best place to add/remove properties from the CO's list is in the setter for Location.
You can't do that if the CO's list is a set of coordinates (well, I suppose you could, but you'd then still be stuck with storing those coordinates in Location)

Reply:
As discussed elsewhere, I think I'm fine with this approach. It does create some data repetition, but does so in a controlled way, and we have that pattern elsewhere already.